src/components/favorites-page/FavoritePage.jsx

```diff
@@ -1,26 +1,25 @@
 import React from "react";
-import PropTypes from "prop-types";
 import {Link} from "react-router-dom";
 import {
-  useSelector,
   // TODO: Нужна для тестов, удалить когда будет сдача проекта
-  // useDispatch,
+  useDispatch,
 } from "react-redux";
 
-import {OfferCardTypes} from "Project/prop-types/offer-card";
 import {
-  selectors,
   // TODO: Нужна для тестов, удалить когда будет сдача проекта
-  // thunks,
+  thunks,
 } from "Project/redux/auth";
 
 import FavoriteCard from "./favorite-card/favorites-page-favorite-card";
+import useFavoritePage from "./hooks/use-favorite-page";
+import Empty from "./favorite-page-empty/favorite-page-empty";
 
-const FavoritesPage = (props) => {
-  const {items} = props;
-  const email = useSelector((state) => selectors.getUserData(state, `email`));
+const FavoritesPage = () => {
   // TODO: Нужна для тестов, удалить когда будет сдача проекта
-  // const dispatch = useDispatch();
+  const dispatch = useDispatch();
+  const {email, cities, offersCardsMap, offersIdsMap, handleFavoriteChange} = useFavoritePage();
+  const hasCities = Boolean(cities) && Boolean(cities.length);
```

Comment:
оператор && приведет к неявному тайпкастингу, нет необходимости создавать экземпляр типа Boolean.

```suggestion
  const hasCities = cities && cities.length;
```

Reply:
а в таком варианте `cities && cities.length;` получится же то что в переменную `hasCities` может записаться число равное длине массива, например если массив с одним элементом то там будет 1, если пустой массив то 0, но суть в том, что переменная называется булевым значением (есть города или нет), но в какой то момент в нем будет boolean, а в другой number